tap_slack/streams.py

```diff
@@ -32,9 +32,23 @@ def get_url_params(self, context, next_page_token):
     def post_process(self, row, context):
         "Join the channel if not a member, but emit no data."
         row = super().post_process(row, context)
-        if not row["is_member"]:
-            self._join_channel(row["id"])
-        return row
+        # return all in selected_channels or default to all, exclude any in excluded_channels list
+        channel_id = row["id"]
+        if self._is_channel_included(channel_id):
+            if not row["is_member"] and self.config.get("auto_join_channels", False):
+                self._join_channel(channel_id)
+            return row
+
+    def _is_channel_included(self, channel_id: str) -> bool:
+        selected_channels = self.config.get("selected_channels")
+        if not selected_channels or channel_id in selected_channels:
```

Comment:
@pnadolny13 The code looks good but we We might be able to refactor this to be a little more transparent:

```
    def _is_channel_included(self, channel_id: str) -> bool:
        selected_channels = self.config.get("selected_channels")
        excluded_channels = self.config.get("excluded_channels", [])
        if channel_id in excluded_channels:
                return False
        if selected_channels and channel_id not in selected_channels:
                return False
        return True
```

Reply:
@stkbailey beautiful! I updated the MR.